### ISSUE TITLE
Hound destroying existing map markers

### ISSUE THREAD

**Issue by delenda-delenda** [uriba107/DCS-Hound#28]
Map markers that are created by other scripts in a mission are being removed when Hound starts drawing.

Increasing _MarkId in HoundUtils to 1 above the already existing number of marks solves the issue.

Would it be possible to either:

1) Use something like world.getMarkPanels( ) to get the number/max index of existing marks so that Hound can adjust its starting ID accordingly
2) Make _MarkId easier to change by making it more accessible (setter function maybe?) and adding documentation

**Comment by uriba107** [comment]
That is interesting.
Can you elaborate a little bit on the markers that are blown away? I assume they are mission specific markers you create via scripting?

The reason I'm asking is because I'm using MOOSE or MIST marker ids where possible to avoid such conflicts.
In addition to that, Hound is tracking the markers it is adding, and deletes its own markers. so it will not delete markers using the same Id Generator (like MOOSE tasking markers)

Player placed markers ID start in the 2,000,000 range IIRC, so they are irrelevant.

The funny thing is that Hound originally used `getMarkPanels` and incremented the ID, however manual markers do not honor this and hell would break loose (Like CTD for Viggen players). 

Anyway, Hound logic on marker IDs is as followes:
```
if MOOSE present, use `UTILS.GetMarkID()`
else if MIST version > 4.5, use `mist.marker.getNextId()`
else increment internal counter.
```

Assuming I guessed correctly about the scenario we are facing, there are multiple options to fix this. 
1. Use `UTILS.GetMarkID()`/`mist.marker.getNextId()` in your code to get a consistent Marker ID
2. Use `HoundUtils.getMarkId()` in your own code to make sure it's consistent numbering
3. When initilizing Hound (i.e after including the script) just override the initial marker ID. 
```lua
HoundUtils._MarkId = 100000
```
choose something out of the way like 100k in this example

I'm marking as "not a bug" for now, waiting to hear from you.

**Comment by delenda-delenda** [comment]
I suppose the bug here is that without that variable being changed, markers get overwritten, and there's nothing in the documentation to suggest that _MarkId should be changed to stop that from happening. 

In this case, it's somebody else's mission that I'm adding Hound to that already has a method of controlling its own markers.

I suspect that the majority of use cases for Hound would be as a "drop-in" ELINT package, so getting mission designers/maintainers to use either of the first two of your options might not be friendliest approach.

I think my option 2 is basically your option 3 - i.e changing _MarkId. I was suggesting adding documentation to the `Map Markers` section in your README to make it clear to mission designers that this would need to be done, otherwise you risk losing your own markers.

Great news on the setter function - this combined with updated documentation should make this issue go away.

**Comment by uriba107** [comment]
You need to take into account that MOOSE will also have the same effect on your scripts. Which is why I'm trying to use it if possible.

Please note that if an external source of markIDs exists in the mission (MOOSE/MIST), the internal counter is not used and ids will start from 1. Becaus that what MOOSE does.

I think I will update the internal mechanism to start from 100k or something just in case.

**Comment by uriba107** [comment]
0.2.2 will allow mission builders to force use the internal counter rather then using the default marker Id allocation logic.

**Comment by uriba107** [comment]
more options shipped with 0.2.2.
closing for now
https://github.com/uriba107/DCS-Hound#living-with-other-marker-spawning-scripts
